packages/integrations/useCookies/index.ts

```diff
@@ -0,0 +1,103 @@
+import { ref, getCurrentInstance, watchEffect } from 'vue-demi'
+import Cookie from 'universal-cookie'
+import type { IncomingMessage } from 'http'
+
+type RawCookies = {
+  [name: string]: string
+}
+
+let universalCookie: Cookie | undefined
+
+/**
+ * Initializing universal-cookie using request (default is window.document.cookie)
+ * @param {Object} req - incoming http request (for SSR)
+ * @see {@link https://github.com/reactivestack/cookies/tree/master/packages/universal-cookie|universal-cookie}
+ */
+export const createCookies = (req?: IncomingMessage) => {
+  universalCookie = new Cookie(req ? req.headers.cookie : null)
+
+  return universalCookie
+}
+
+/**
+ * Reactive methods to work with cookies (firstly call {@link createCookies} method if using SSR)
+ * @param {string[]|null|undefined} dependencies - array of watching cookie's names. Pass empty array if don't want to watch cookies changes.
+ * @param {boolean} [doNotParse=false] - don't try parse value as JSON
+ */
+export const useCookies = (dependencies?: string[] | null, doNotParse = false) => {
+  const cookies = universalCookie || createCookies()
+
+  let previousCookies = cookies.getAll<RawCookies>({ doNotParse: true })
+
+  /**
+   * Adds reactivity to returning methods
+   */
+  const touches = ref(0)
+
+  /**
+   * Calls inside setup() function only
+   */
+  if (getCurrentInstance()) {
+    const onChange = () => {
+      const newCookies = cookies.getAll({ doNotParse: true })
+
+      if (
+        shouldUpdate(
+          dependencies || null,
+          newCookies,
+          previousCookies,
+        )
+      )
+        touches.value++
+
+      previousCookies = newCookies
+    }
+
+    watchEffect(() => {
+      cookies.addChangeListener(onChange)
+
+      return () => {
+        cookies.removeChangeListener(onChange)
+      }
+    })
+  }
+
+  return {
```

Comment:
Would it be nice to also export a reactive cookies state?

Thinking:

```ts
const cookies = computed<T>(() => {
  touches.value
  return _cookies.getAll<T>({ doNotParse })
})
```

Reply:
I think (1)
```vue
{{ cookies.get('locale') }}
```
is better than (2)
```vue
{{ cookies.getAll().locale }}
or
{{ cookies.value.locale }}
```

Because honestly you will never use `cookies.getAll()` inside your application (only for debugging purposes)
And (1) example ... em ... reminds (?) you don't forget provide dependencies array :)

I also think about automatically adding new cookie names to dependencies when someone tried to get its value. With your example it won't be possible.